source/DomainTheory/Topology/ScottTopologyProperties.lagda

```diff
@@ -0,0 +1,225 @@
+
+\end{code}
+
+Conversely, if the principal filter is Scott open then `c` is a compact element.
+
+\begin{code}
+
+ principal-filter-scott-open-implies-compact : (c : ⟨ 𝓓 ⟩)
+                                             → is-scott-open (↑[ 𝓓 ] c) holds
+                                             → is-compact 𝓓 c
+ principal-filter-scott-open-implies-compact c (υ , κ) I ι δ p =
+  κ ((I , ι) , δ) p
+
+\end{code}
+
+We can now record this as a logical equivalence.
+
+\begin{code}
+
+ principal-filter-scott-open-iff-compact :
+  (x : ⟨ 𝓓 ⟩) → is-scott-open (↑[ 𝓓 ] x) holds ↔ is-compact 𝓓 x
+ principal-filter-scott-open-iff-compact x = Ⅰ , Ⅱ
+  where
+   Ⅰ = principal-filter-scott-open-implies-compact x
+   Ⅱ = compact-implies-principal-filter-is-scott-open x
+
+\end{code}
+
+Notation for the principal Scott open.
+
+\begin{code}
+
+ ↑ˢ[_] : (Σ c ꞉ ⟨ 𝓓 ⟩ , is-compact 𝓓 c) → Σ S ꞉ 𝓟 {𝓥} ⟨ 𝓓 ⟩ , is-scott-open S holds
+ ↑ˢ[ (c , κ) ] =
+  principal-filter 𝓓 c , compact-implies-principal-filter-is-scott-open c κ
+
+\end{code}
+
+\begin{code}
+
+module PropertiesAlgebraic (𝓓 : DCPO {𝓤} {𝓥})
+                           (𝕒 : structurally-algebraic 𝓓) where
+
+ open DefnOfScottTopology 𝓓 𝓥
+
+ open structurally-algebraic
+
+ is-compactₚ : ⟨ 𝓓 ⟩ → Ω (𝓤 ⊔ 𝓥 ⁺)
+ is-compactₚ x = is-compact 𝓓 x , being-compact-is-prop 𝓓 x
+
+ join-of-compact-opens : 𝓟 {𝓥} ⟨ 𝓓 ⟩ → 𝓟 {𝓤 ⊔ 𝓥 ⁺} ⟨ 𝓓 ⟩
+ join-of-compact-opens U x =
+  Ǝ c ꞉ ⟨ 𝓓 ⟩ , (is-compactₚ c ∧ c ∈ₚ U ∧ x ∈ₚ (↑[ 𝓓 ] c)) holds
+
+ characterization-of-scott-opens₁ : (U : 𝓟 ⟨ 𝓓 ⟩)
+                                  → is-scott-open U holds
+                                  → U ⊆ join-of-compact-opens U
+ characterization-of-scott-opens₁ U (υ , ξ) x p = †
+  where
+   S : Fam 𝓥 ⟨ 𝓓 ⟩
+   S = index-of-compact-family 𝕒 x , compact-family 𝕒 x
+
+   S↑ : Fam↑
+   S↑ = S , compact-family-is-directed 𝕒 x
+
+   q : x ＝ ⋁ S↑
+   q = compact-family-∐-＝ 𝕒 x ⁻¹
+
+   κ : (i : index S) → is-compactₚ (S [ i ]) holds
+   κ = compact-family-is-compact 𝕒 x
+
+   ψ : is-upperbound (underlying-order 𝓓) x (S [_])
+   ψ i = transport (λ - → (S [ i ]) ⊑⟨ 𝓓 ⟩ -) (q ⁻¹) (⋁-is-upperbound S↑ i)
+
+   φ : (⋁ S↑) ∈ U
+   φ = transport (λ - → - ∈ U) q p
+
+   ‡ : Σ i ꞉ index S , (S [ i ]) ∈ U
+     → ∃ c₀ ꞉ ⟨ 𝓓 ⟩ , (is-compactₚ c₀ ∧ c₀ ∈ₚ U ∧ x ∈ₚ ↑[ 𝓓 ] c₀) holds
+   ‡ (i , μ) = ∣ S [ i ] , κ i , μ , ψ i ∣
+
+   † : ∃ c₀ ꞉ ⟨ 𝓓 ⟩ , (is-compactₚ c₀ ∧ c₀ ∈ₚ U ∧ x ∈ₚ ↑[ 𝓓 ] c₀) holds
+   † = ∥∥-rec ∃-is-prop ‡ (ξ S↑ φ)
+
+ characterization-of-scott-opens₂ : (U : 𝓟 ⟨ 𝓓 ⟩)
+                                  → is-scott-open U holds
+                                  → join-of-compact-opens U ⊆ U
+ characterization-of-scott-opens₂ U (υ , _) x p =
+  ∥∥-rec (holds-is-prop (x ∈ₚ U)) † p
+   where
+    † : Σ c ꞉ ⟨ 𝓓 ⟩ , (is-compactₚ c ∧ c ∈ₚ U ∧ principal-filter 𝓓 c x) holds
+      → x ∈ₚ U holds
+    † (c , _ , q , r) = υ c x q r
+
+\end{code}
+
+\begin{code}
+
+ characterization-of-scott-opens : (U : 𝓟 {𝓥} ⟨ 𝓓 ⟩)
```

Comment:
Maybe write this as follows for the sake of legibility:
```
characterization-of-scott-opens 
 : (U : 𝓟 {𝓥} ⟨ 𝓓 ⟩)
→ (is-scott-open U ⇒ (Ɐ x ꞉ ⟨ 𝓓 ⟩ , U x ⇔ join-of-compact-opens U x)) holds
```

Reply:
Good idea. Done now.